autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -96,24 +112,27 @@ def _fit(
         # The final output of a validator is a numpy array. But pandas
         # gives us information about the column dtype
         if isinstance(X, np.ndarray):
+
             X = self.numpy_array_to_pandas(X)
+            # Replace the data type from the previously saved type.
+            self.data_type = type(X)
+            # save all the information about the column order and data types
+            self._check_data(X)
```

Comment:
Hmm, well this is a bit complex. `_check_data ` is already called in `base_feature_validator`. If you pass a numpy array there, it will have a numpy data type, while above, it is converted to DataFrame, when you call` _check_data` again you will have a warning in the logger that the data type was changed, a warning that we trigger. I do not think it fits `_check_data` to return a type, the idea is to just verify that the data type is the same and the column order and types are the same. The problem here is that we convert it ourselves.